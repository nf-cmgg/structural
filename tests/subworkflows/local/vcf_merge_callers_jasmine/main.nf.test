nextflow_workflow {

    name "Test Workflow VCF_MERGE_CALLERS_JASMINE"
    script "subworkflows/local/vcf_merge_callers_jasmine/main.nf"
    workflow "VCF_MERGE_CALLERS_JASMINE"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "vcf_merge_callers_jasmine"

    config "./nextflow.config"

    test("homo_sapiens - multiple callers") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", caller:"delly"],
                    file(params.sv_vcf1, checkIfExists:true)
                ], [
                    [id:"test", sample:"test", caller:"expansionhunter"],
                    file(params.repeats_vcf1, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = ["delly", "expansionhunter"]
                input[4] = "sv"
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path(workflow.out.vcfs[0][1]).linesGzip[-4..-1],
                    file(workflow.out.vcfs[0][2]).name,
                ).match() }
            )
        }

    }

    test("homo_sapiens - one caller") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", caller:"delly"],
                    file(params.sv_vcf1, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = ["delly"]
                input[4] = "sv"
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    file(workflow.out.vcfs[0][1]).name,
                    file(workflow.out.vcfs[0][2]).name,
                ).match() }
            )
        }

    }

}
