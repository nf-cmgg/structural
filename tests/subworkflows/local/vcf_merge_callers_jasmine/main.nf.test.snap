{
    "homo_sapiens - multiple callers": {
        "content": [
            [
                "chr16\t1061455\t1_1_manta_DEL_1\tGTTCCACTCATGTTGTTGCAGATTACTGGATCTCACTCTTTTTTATGGGTGAATCGTACTTCATAGTGTATATGTACC\tG\t619\tPASS\tCALLERS=manta;CIPOS=0,5;END=1061532;SVLEN=-77;SVTYPE=DEL;STARTVARIANCE=0.000000;ENDVARIANCE=0.000000;AVG_LEN=-77.000000;AVG_START=1061455.000000;AVG_END=1061532.000000;VARCALLS=1;ALLVARS_EXT=((manta_DEL_1));SUPP_VEC_EXT=0010;IDLIST_EXT=manta_DEL_1;SUPP_EXT=1;SUPP_VEC=01;SUPP=1;SVMETHOD=JASMINE;IDLIST=1_manta_DEL_1;INTRASAMPLE_IDLIST=1_manta_DEL_1\tGT:PE:SR\t1/1:0,0:0,19",
                "chr16\t1077371\t1_0_delly_INV_2\tT\t<INV>\t58\tLowQual\tCALLERS=delly;CIEND=-392,392;CIPOS=-392,392;END=1078502;SVLEN=1131;SVTYPE=INV;STARTVARIANCE=0.000000;ENDVARIANCE=0.000000;AVG_LEN=1131.000000;AVG_START=1077371.000000;AVG_END=1078502.000000;VARCALLS=1;ALLVARS_EXT=((delly_INV_2));SUPP_VEC_EXT=0100;IDLIST_EXT=delly_INV_2;SUPP_EXT=1;SUPP_VEC=01;SUPP=1;SVMETHOD=JASMINE;IDLIST=0_delly_INV_2;INTRASAMPLE_IDLIST=0_delly_INV_2\tGT:PE:SR\t0/0:19,2:0,0",
                "chr16\t1123476\t1_0_delly_INV_3\tA\t<INV>\t180\tPASS\tCALLERS=delly;CIEND=-3,3;CIPOS=-3,3;END=1604486;SVLEN=481010;SVTYPE=INV;STARTVARIANCE=0.000000;ENDVARIANCE=0.000000;AVG_LEN=481010.000000;AVG_START=1123476.000000;AVG_END=1604486.000000;VARCALLS=1;ALLVARS_EXT=((delly_INV_3));SUPP_VEC_EXT=0100;IDLIST_EXT=delly_INV_3;SUPP_EXT=1;SUPP_VEC=01;SUPP=1;SVMETHOD=JASMINE;IDLIST=0_delly_INV_3;INTRASAMPLE_IDLIST=0_delly_INV_3\tGT:PE:SR\t1/1:0,0:0,12",
                "chr16\t1135261\t1_0_delly_INS_4\tC\t<INS>\t299\tPASS\tCALLERS=delly;CIEND=-2,2;CIPOS=-2,2;END=1135261;SVLEN=27;SVTYPE=INS;STARTVARIANCE=0.000000;ENDVARIANCE=0.000000;AVG_LEN=27.000000;AVG_START=1135261.000000;AVG_END=1135261.000000;VARCALLS=1;ALLVARS_EXT=((delly_INS_4));SUPP_VEC_EXT=0100;IDLIST_EXT=delly_INS_4;SUPP_EXT=1;SUPP_VEC=01;SUPP=1;SVMETHOD=JASMINE;IDLIST=0_delly_INS_4;INTRASAMPLE_IDLIST=0_delly_INS_4\tGT:PE:SR\t1/1:0,0:0,25"
            ],
            "test.sv.vcf.gz.tbi",
            5,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-11T14:40:37.603583687"
    },
    "homo_sapiens - one caller": {
        "content": [
            "test.sv.vcf.gz",
            "test.sv.vcf.gz.tbi",
            5,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-11T14:40:59.466087604"
    }
}